### DISCUSSION TITLE
Custom parameter info

### DISCUSSION THREAD

**Discussion by noman-khan1224** [eclipse-langium/langium#1511] [Q&A]
I want to fetch data, and I use the updateContent method for this purpose. However, within this method, there is a for loop that assigns an empty value to the node. If I remove this for loop from the updateContent method, will it cause any performance issues? Also, what is the purpose of this for loop?

![Screenshot (8)](https://github.com/eclipse-langium/langium/assets/161299971/11d66745-f77b-495e-bf76-afb1f322efda)

Request you to help with an approach for same?

**Comment by msujew** [comment]
Hey @noman-khan1224,

you seem to be using an outdated version of Langium. We no longer clear the exported node of an `AstNodeDescription`:

https://github.com/eclipse-langium/langium/blob/6971c83afb924336d70b3e3092440478c73ac92e/packages/langium/src/workspace/index-manager.ts#L154-L160

The purpose used to be to reduce our memory footprint by deleting references to elements that have been potentially unloaded from the workspace. However, we no longer support that use case and so we also deleted the loop.
